cmake/modules/Buildpmem.cmake

```diff
@@ -47,7 +47,7 @@ function(build_pmem)
   set_target_properties(pmem::pmem PROPERTIES
     INTERFACE_INCLUDE_DIRECTORIES ${PMDK_INCLUDE}
     IMPORTED_LOCATION "${PMDK_LIB}/libpmem.a"
-    INTERFACE_LINK_LIBRARIES ${CMAKE_THREAD_LIBS_INIT})
```

Comment:
@ktdreyer could you also add `find_package(Threads)` right before the `set_target_properties()` call. so it's more correct. as per https://cmake.org/cmake/help/latest/module/FindThreads.html, `CMAKE_THREAD_LIBS_INIT` is set by this module.

Reply:
If we are requiring cmake 3.10, perhaps we should consider something along the lines of https://github.com/glfw/glfw/commit/49c5d837b4667e2c6e09a54f4141fc744af0558f?  Threads::Threads is available since 3.1.

Reply:
agreed, more modernized this way.

Reply:
I'm sorry I'm lost - you'll have to spell out exactly what change you want me to do here :laughing:

Reply:
something like this
```diff
    cmake: link Threads::Threads instead of CMAKE_THREAD_LIBS_INIT

    `CMAKE_THREAD_LIBS_INIT` can be empty. On CentOS Stream 9, this leads to an error:

    ```
      CMake Error at cmake/modules/Buildpmem.cmake:47 (set_target_properties):
        set_target_properties called with incorrect number of arguments.
    ```

    because this is because glibc 2.34 merged libpthread.so into libc.so.
    and fedora35 and el9 comes with glibc 2.34.

    Fixes: https://tracker.ceph.com/issues/52353
    Signed-off-by: Kefu Chai <kchai@redhat.com>

diff --git a/cmake/modules/Buildpmem.cmake b/cmake/modules/Buildpmem.cmake
index 6db9754cb95..ead5c80ae90 100644
--- a/cmake/modules/Buildpmem.cmake
+++ b/cmake/modules/Buildpmem.cmake
@@ -44,10 +44,11 @@ function(build_pmem)
   add_library(pmem::pmem STATIC IMPORTED GLOBAL)
   add_dependencies(pmem::pmem pmdk_ext)
   file(MAKE_DIRECTORY ${PMDK_INCLUDE})
+  find_package(Threads)
   set_target_properties(pmem::pmem PROPERTIES
     INTERFACE_INCLUDE_DIRECTORIES ${PMDK_INCLUDE}
     IMPORTED_LOCATION "${PMDK_LIB}/libpmem.a"
-    INTERFACE_LINK_LIBRARIES ${CMAKE_THREAD_LIBS_INIT})
+    INTERFACE_LINK_LIBRARIES Threads::Threads)

   # libpmemobj
   add_library(pmem::pmemobj STATIC IMPORTED GLOBAL)
```